orchestration/cluster_test.go

```diff
@@ -0,0 +1,221 @@
+		ClusterArn:          aws.String("arn:aws:ecs:us-east-1:1234567890:cluster/cluster2"),
+		ClusterName:         aws.String("cluster2"),
+		DefaultCapacityProviderStrategy: []*ecs.CapacityProviderStrategyItem{
+			&ecs.CapacityProviderStrategyItem{
+				Base:             aws.Int64(1),
+				CapacityProvider: aws.String("FARGATE"),
+				Weight:           aws.Int64(0),
+			},
+			&ecs.CapacityProviderStrategyItem{
+				CapacityProvider: aws.String("FARGATE_SPOT"),
+				Weight:           aws.Int64(1),
+			},
+		},
+		PendingTasksCount:                 aws.Int64(1),
+		RegisteredContainerInstancesCount: aws.Int64(1),
+		RunningTasksCount:                 aws.Int64(1),
+		Status:                            aws.String("ACTIVE"),
+		Tags: []*ecs.Tag{
+			&ecs.Tag{
+				Key:   aws.String("fuz"),
+				Value: aws.String("biz"),
+			},
+		},
+	},
+	&ecs.Cluster{
+		ActiveServicesCount: aws.Int64(3),
+		CapacityProviders: []*string{
+			aws.String("FARGATE"),
+			aws.String("FARGATE_SPOT"),
+		},
+		ClusterArn:  aws.String("arn:aws:ecs:us-east-1:1234567890:cluster/cluster3"),
+		ClusterName: aws.String("cluster3"),
+		DefaultCapacityProviderStrategy: []*ecs.CapacityProviderStrategyItem{
+			&ecs.CapacityProviderStrategyItem{
+				Base:             aws.Int64(1),
+				CapacityProvider: aws.String("FARGATE"),
+				Weight:           aws.Int64(0),
+			},
+			&ecs.CapacityProviderStrategyItem{
+				CapacityProvider: aws.String("FARGATE_SPOT"),
+				Weight:           aws.Int64(1),
+			},
+		},
+		PendingTasksCount:                 aws.Int64(1),
+		RegisteredContainerInstancesCount: aws.Int64(1),
+		RunningTasksCount:                 aws.Int64(1),
+		Status:                            aws.String("ACTIVE"),
+		Tags: []*ecs.Tag{
+			&ecs.Tag{
+				Key:   aws.String("foo"),
+				Value: aws.String("bar"),
+			},
+		},
+	},
+}
+
+func (m *mockECSClient) CreateClusterWithContext(ctx context.Context, input *ecs.CreateClusterInput, opts ...request.Option) (*ecs.CreateClusterOutput, error) {
+	if m.err != nil {
+		return nil, m.err
+	}
+
+	for _, cluster := range testClusters {
+		if aws.StringValue(input.ClusterName) == aws.StringValue(cluster.ClusterName) {
+			return &ecs.CreateClusterOutput{
+				Cluster: cluster,
+			}, nil
+		}
+
+	}
+
+	return nil, errors.New("boom!")
+}
+
+func (m *mockECSClient) DescribeClustersWithContext(ctx aws.Context, input *ecs.DescribeClustersInput, opts ...request.Option) (*ecs.DescribeClustersOutput, error) {
+	if m.err != nil {
+		return nil, m.err
+	}
+
+	clusters := []*ecs.Cluster{}
+	for _, inputClu := range input.Clusters {
+		for _, cluster := range testClusters {
+			if aws.StringValue(inputClu) == aws.StringValue(cluster.ClusterName) {
+				clusters = append(clusters, cluster)
+			}
+
+		}
+	}
+
+	return &ecs.DescribeClustersOutput{Clusters: clusters}, nil
+}
+
+func TestProcessCluster(t *testing.T) {
+	orchestrator := newMockOrchestrator(t, "myorg", nil, nil, nil, nil, nil)
```

Comment:
This would've been nicer if Go supported default arg values 😆

Reply:
it accepts variable length args and the function option pattern, but I was too lazy to do that 😆